### PR TITLE
Remove send mail option

### PR DESCRIPTION
**Pull request by schiessle**
Remove remaining "notify by email" and "send link by mail" code. Was already started here, but there was still some code left: https://github.com/nextcloud/server/pull/1929

As discussed here: https://github.com/nextcloud/server/issues/1916

cc @rullzer

### CODE REVIEW

File: apps/files/lib/Controller/ViewController.php [hunk 1]
```diff
@@ -235,7 +235,6 @@ public function index($dir = '', $view = '', $fileid = null, $fileNotFound = fal
 		$params['owner'] = $storageInfo['owner'];
 		$params['ownerDisplayName'] = $storageInfo['ownerDisplayName'];
 		$params['isPublic'] = false;
-		$params['mailNotificationEnabled'] = $this->config->getAppValue('core', 'shareapi_allow_mail_notification', 'no');
 		$params['mailPublicNotificationEnabled'] = $this->config->getAppValue('core', 'shareapi_allow_public_notification', 'no');
```
Comment (nickvergessen) [inline apps/files/lib/Controller/ViewController.php]: Remove this as well?

File: apps/files/templates/index.php [hunk 2]
```diff
@@ -16,7 +16,6 @@
 <input type="hidden" name="ownerDisplayName" id="ownerDisplayName" value="<?php p($_['ownerDisplayName']); ?>" />
 <input type="hidden" name="fileNotFound" id="fileNotFound" value="<?php p($_['fileNotFound']); ?>" />
 <?php if (!$_['isPublic']) :?>
-<input type="hidden" name="mailNotificationEnabled" id="mailNotificationEnabled" value="<?php p($_['mailNotificationEnabled']) ?>" />
 <input type="hidden" name="mailPublicNotificationEnabled" id="mailPublicNotificationEnabled" value="<?php p($_['mailPublicNotificationEnabled']) ?>" />
```
Comment (nickvergessen) [inline apps/files/templates/index.php]: Remove this as well?

File: apps/files/tests/Controller/ViewControllerTest.php [hunk 3]
```diff
@@ -218,7 +218,6 @@ public function testIndexWithRegularBrowser() {
 				'defaultFileSortingDirection' => 'asc',
 				'showHiddenFiles' => 0,
 				'fileNotFound' => 0,
-				'mailNotificationEnabled' => 'no',
 				'mailPublicNotificationEnabled' => 'no',
```
Comment (nickvergessen) [inline apps/files/tests/Controller/ViewControllerTest.php]: Remove this as well?

File: apps/files_sharing/lib/Capabilities.php [hunk 4]
```diff
@@ -73,8 +73,6 @@ public function getCapabilities() {
 			}
 			$res["public"] = $public;
 
-			$res['user']['send_mail'] = $this->config->getAppValue('core', 'shareapi_allow_mail_notification', 'no') === 'yes';
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Capabilities.php]: set to false, so a check for the capability on client is saver?

File: apps/files_sharing/lib/Migration.php [hunk 5]
```diff
@@ -111,6 +116,13 @@ public function updateInitiatorInfo() {
 	}
 
 	/**
+	 * this was dropped for Nextcloud 11 in favour of share by mail
+	 */
+	public function removeSendMailOption() {
+		$this->config->deleteAppValue('core', 'shareapi_allow_mail_notification');
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Migration.php]: Also `shareapi_allow_public_notification`

File: core/js/shareconfigmodel.js [hunk 6]
```diff
@@ -56,13 +56,6 @@
 		/**
 		 * @returns {boolean}
 		 */
-		isMailNotificationEnabled: function() {
```
Comment (nickvergessen) [inline core/js/shareconfigmodel.js]: Also `isMailPublicNotificationEnabled`

File: lib/private/Settings/Admin/Sharing.php [hunk 7]
```diff
@@ -52,7 +52,6 @@ public function getForm() {
 			// Built-In Sharing
 			'allowGroupSharing'               => $this->config->getAppValue('core', 'shareapi_allow_group_sharing', 'yes'),
 			'allowLinks'                      => $this->config->getAppValue('core', 'shareapi_allow_links', 'yes'),
-			'allowMailNotification'           => $this->config->getAppValue('core', 'shareapi_allow_mail_notification', 'no'),
 			'allowPublicMailNotification'     => $this->config->getAppValue('core', 'shareapi_allow_public_notification', 'no'),
```
Comment (nickvergessen) [inline lib/private/Settings/Admin/Sharing.php]: drop

File: tests/lib/Settings/Admin/SharingTest.php [hunk 8]
```diff
@@ -62,55 +62,50 @@ public function testGetFormWithoutExcludedGroups() {
 		$this->config
 			->expects($this->at(3))
 			->method('getAppValue')
-			->with('core', 'shareapi_allow_mail_notification', 'no')
-			->willReturn('no');
-		$this->config
-			->expects($this->at(4))
-			->method('getAppValue')
 			->with('core', 'shareapi_allow_public_notification', 'no')
```
Comment (nickvergessen) [inline tests/lib/Settings/Admin/SharingTest.php]: drop

File: tests/lib/Settings/Admin/SharingTest.php [hunk 9]
```diff
@@ -161,55 +155,50 @@ public function testGetFormWithExcludedGroups() {
...
```
Comment (nickvergessen) [inline tests/lib/Settings/Admin/SharingTest.php]: drop

File: tests/lib/Settings/Admin/SharingTest.php [hunk 10]
```diff
@@ -121,7 +116,6 @@ public function testGetFormWithoutExcludedGroups() {
 			[
 				'allowGroupSharing'               => 'yes',
 				'allowLinks'                      => 'yes',
-				'allowMailNotification'           => 'no',
 				'allowPublicMailNotification'     => 'no',
```
Comment (nickvergessen) [inline tests/lib/Settings/Admin/SharingTest.php]: drop

File: tests/lib/Settings/Admin/SharingTest.php [hunk 11]
```diff
@@ -220,7 +209,6 @@ public function testGetFormWithExcludedGroups() {
...
```
Comment (nickvergessen) [inline tests/lib/Settings/Admin/SharingTest.php]: drop

**Review by nickvergessen — DISMISSED** [review]

File: apps/files_sharing/lib/Capabilities.php (hunk 4 above)
Comment (rullzer) [inline apps/files_sharing/lib/Capabilities.php]: Agreed

Comment (schiessle) [inline apps/files_sharing/lib/Capabilities.php]: done

File: apps/files_sharing/lib/Migration.php (hunk 5 above)
Comment (schiessle) [inline apps/files_sharing/lib/Migration.php]: done

**Comment by schiessle** [comment]
@rullzer @nickvergessen please review... everything should be fine now

**Comment by rullzer** [comment]
good stuff! :+1: 

I updated `core.json`

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Comment by MorrisJobke** [comment]
Slightly different, but anyways: https://github.com/nextcloud/server/pull/5897

**Pull request merged**